src/Storages/ProjectionsDescription.cpp

```diff
@@ -217,23 +211,12 @@ ProjectionDescription::getMinMaxCountProjection(const ColumnsDescription & colum
         result.query_ast, query_context, storage, {}, SelectQueryOptions{QueryProcessingStage::WithMergeableState}.modify().ignoreAlias());
     result.required_columns = select.getRequiredColumns();
     result.sample_block = select.getSampleBlock();
-
-    for (size_t i = 0; i < result.sample_block.columns(); ++i)
-    {
-        const auto & column_with_type_name = result.sample_block.getByPosition(i);
-
-        if (column_with_type_name.column && isColumnConst(*column_with_type_name.column))
-            throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Projections cannot contain constant columns: {}", column_with_type_name.name);
-
-        result.column_names.emplace_back(column_with_type_name.name);
-        result.data_types.emplace_back(column_with_type_name.type);
-    }
     result.type = ProjectionDescription::Type::Aggregate;
     StorageInMemoryMetadata metadata;
     metadata.setColumns(ColumnsDescription(result.sample_block.getNamesAndTypesList()));
-    metadata.partition_key = KeyDescription::getSortingKeyFromAST({}, metadata.columns, query_context, {});
-    metadata.sorting_key = KeyDescription::getSortingKeyFromAST({}, metadata.columns, query_context, {});
-    metadata.primary_key = KeyDescription::getKeyFromAST({}, metadata.columns, query_context);
+    metadata.partition_key = KeyDescription::getSortingKeyFromAST({}, {}, query_context, {});
+    metadata.sorting_key = KeyDescription::getSortingKeyFromAST({}, {}, query_context, {});
+    metadata.primary_key = KeyDescription::getKeyFromAST({}, {}, query_context);
```

Comment:
It is confusing that those functions are called, well, on nothing.
Maybe default constructor for KeyDescription will do? (or maybe add one from context)

Reply:
Indeed. I'll make a proper constructor for this.